### PR TITLE
Read and parse good_redirects line by line rather than all at once.

### PR DESCRIPTION
**Pull request by IhmeHippi**
Also added unit tests for good_redirects parsing.

### CODE REVIEW

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886&bid=20068353)

File: SwedbankPaySDKTests/FileLinesTests.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class FileLinesTests: XCTestCase {
+    private func checkThatFile(with content: String, parsesTo lines: String...) throws {
+        let file = try FileUtils.openString(content)
+        defer {
+            fclose(file)
+        }
+        file.getLines().assertElementsEqual(lines)
+    }
+
```
Comment (payex-dev) [inline SwedbankPaySDKTests/FileLinesTests.swift:13]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886)

File: SwedbankPaySDKTests/Utils/FileUtils.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        try content.withUnsafeBytes {
+            let size = $0.count
+            XCTAssert(size > 0)
+            let file = try XCTUnwrap(fmemopen(nil, size, "rb+"))
+            XCTAssertEqual(fwrite($0.baseAddress, size, 1, file), 1)
+            rewind(file)
+            return file
+        }
+    }
+
```
Comment (payex-dev) [inline SwedbankPaySDKTests/Utils/FileUtils.swift:16]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886)

File: SwedbankPaySDKTests/GoodWebViewRedirectsTests.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    
+    func testEnsemble() {
+        parse("""
+# Comment
+domain.com
+ sub.domain.com
+
+#not.this.com
+*.subdomains.org
```
Comment (payex-dev) [inline SwedbankPaySDKTests/GoodWebViewRedirectsTests.swift:67]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886)

File: SwedbankPaySDKTests/GoodWebViewRedirectsTests.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import XCTest
+import Foundation
+@testable import SwedbankPaySDK
+
+class GoodWebViewRedirectsTests: XCTestCase {
+    private var redirects: GoodWebViewRedirects!
+
```
Comment (payex-dev) [inline SwedbankPaySDKTests/GoodWebViewRedirectsTests.swift:7]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886)

File: SwedbankPaySDK/Classes/Utils/FileLines.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+struct FileLines: Sequence, IteratorProtocol {
+    private let file: UnsafeMutablePointer<FILE>
+
```
Comment (payex-dev) [inline SwedbankPaySDK/Classes/Utils/FileLines.swift:20]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6137886)

File: SwedbankPaySDKTests/FileLinesTests.swift (hunk 1 above)
Comment (IhmeHippi) [reply SwedbankPaySDKTests/FileLinesTests.swift:13]: Codacy is being weird here. These lines are identical to other single blank lines in the same files. Ignoring these for now.

**Pull request merged**